### PR TITLE
Decouple moving collections into public descriptive from ActiveFedora

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔

Ref #3098

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: app/services/publish/public_desc_metadata_service.rb [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
   # Creates the descriptive XML that we display on purl.stanford.edu
   class PublicDescMetadataService
-    attr_reader :object
+    attr_reader :object, :cocina_object
 
     NOKOGIRI_DEEP_COPY = 1
     MODS_NS = 'http://www.loc.gov/mods/v3'
 
-    def initialize(object)
+    def initialize(object, cocina_object)
```
Comment (justinlittman) [inline app/services/publish/public_desc_metadata_service.rb:11]: Can you rename `object`? Not at all clear what type of object that is.

File: app/controllers/metadata_controller.rb [hunk 2]
```diff
@@ -2,17 +2,17 @@
 
 # A controller to display derived metadata about an object
 class MetadataController < ApplicationController
-  before_action :load_item, :load_cocina_object
-  before_action :load_cocina_object, only: [:public_xml]
+  before_action :load_item
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/controllers/metadata_controller.rb:5]: I didn't change this, I just removed the second argument.  As far as I know everything still needs it.

File: app/services/publish/public_desc_metadata_service.rb (hunk 1 above)
Comment (jcoyne) [reply app/services/publish/public_desc_metadata_service.rb:11]: If we merge this and finish up https://github.com/sul-dlss/dor-services-app/issues/3098, then that argument will be removed.

File: app/services/collections_query.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Queries about collections
+class CollectionsQuery
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/services/collections_query.rb]: I've moved that method.

**Pull request merged**